viewer-prototype/src/node/trace-server-service.ts

```diff
@@ -4,11 +4,24 @@ import { TraceServerConfigService } from '../common/trace-server-config';
 @injectable()
 export class TraceServerServiceImpl implements TraceServerConfigService {
 
-    async startTraceServer(path: string | undefined, port: number | undefined): Promise<void> {
-        spawn(`${path}`, ['-vmargs', `-Dtraceserver.port=${port}`]);
+    startTraceServer(path: string | undefined, port: number | undefined): Promise<void> {
+        const server = spawn(`${path}`, ['-vmargs', `-Dtraceserver.port=${port}`]);
+        return new Promise((resolve, reject) => {
+            server.on('message', () => resolve());
```

Comment:
As far as I understand, message is only called when there are messages send between parent and child process. I have seen code when doing a google search, that message is used with a fork and not spawn. In my environment the promise never resolves here.

Reply:
I've switched this to listen for output on `stdout` and `stderr` and resolve the promise if output is detected on either channel.